h2o-core/src/main/java/water/ChunkAutoBufferWriter.java

```diff
@@ -0,0 +1,89 @@
+package water;
+
+import water.fvec.Chunk;
+import water.fvec.ChunkUtils;
+import water.fvec.Frame;
+import water.parser.BufferedString;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import static water.ExternalFrameUtils.*;
+import static water.ExternalFrameUtils.EXPECTED_STRING;
+
+public final class ChunkAutoBufferWriter implements Closeable  {
+    
+    private final AutoBuffer buffer;
+    
+    public ChunkAutoBufferWriter(OutputStream outputStream) {
+        this.buffer = new AutoBuffer(outputStream, false);
+    }
+
+    public void writeChunk(String frameName, int chunkId, byte[] expectedTypes, int[] selectedColumnIndices) {
```

Comment:
Should we check the `expectedTypes` really match the types of vectors red from the Frame ? The type mapping has already been done by @jakubhava in `ExternalFrameUtils.vecTypesFromExpectedTypes`.

Reply:
Great idea, thanks!